smack-core/src/main/java/org/jivesoftware/smack/util/StringUtils.java

```diff
@@ -270,7 +270,7 @@ public static String encodeHex(byte[] bytes) {
     /**
      * 24 upper case characters from the latin alphabet and numbers without '0' and 'O'.
      */
-    private static final char[] UNAMBIGUOUS_NUMBERS_AND_LETTER = "123456789ABCDEFGHIJKLMNPQRSTUVWXYZ".toCharArray();
+    public static final char[] UNAMBIGUOUS_NUMBERS_AND_LETTER = "123456789ABCDEFGHIJKLMNPQRSTUVWXYZ".toCharArray();
```

Comment:
If this array is public, then users would modify its contents. I think it should stay private. I guess you idea was to use it in the regex below, which we definitely should!
How about:

```
public static final String UNAMBIGUOUS_NUMBERS_AND_LETTER_STRING = "123456789ABCDEFGHIJKLMNPQRSTUVWXYZ";
private static final char[] UNAMBIGUOUS_NUMBERS_AND_LETTER = UNAMBIGUOUS_NUMBERS_AND_LETTER_STRING.toCharArray();
```

Reply:
Ah right, haven't thought of that.
Yeah I guess either the fix you suggested, or just make it private again, as I ended up constructing the regex without the array.